src/posthog-core.js

```diff
@@ -873,10 +874,12 @@ PostHogLib.prototype.identify = function (new_distinct_id, userPropertiesToSet,
     ) {
         this.capture(
             '$identify',
-            {
-                distinct_id: new_distinct_id,
-                $anon_distinct_id: previous_distinct_id,
-            },
+            this.get_config('send_anon_distinct_id')
```

Comment:
I realise my solution if also prone to error if a call with `previous_distinct_id` happens in a different session to the identify call. this could I guess be stored as a flag in localstorage instead like `last_distinct_id_used`. On identify we check this value  - if empty we don't send the `$anon_distinct_id`, if set and different from the new one, we do send it?

Reply:
Can you implement that in the current codebase 🙃

Reply:
This is implementable in the current codebase with full coverage no problem. The should is the question :)

Writing out the bug, is this something like the following?
1. Enable `posthog-js` on your site with no autocapture/no automatic pageviews
2. Call `.identify` as the first event on the site
3. Call `.identify` a subsequent time later on

Expected result:
1. Single person gets created

Actual result:
1. Two persons exist in total

If so this is an intentional design decision, we don't allow merges A -> B -> C. The key question is why is the user calling `identify` in the first place at step (2)? A common failure mode we've seen is that the user is anonymous but clients are trying to add properties, which should be done elsehow.

Reply:
The plan Ben and I discussed is to convert current codebase to TypeScript, and only then take on the refactor with confidence. I disagree regarding tests and coverage, but that's a longer discussion not worth getting into.

The bug is:

1. Enable posthog-js on your site with no autocapture/no automatic pageviews
2. Call .identify as the first event on the site

As a result, the distinct ID sent via `$anon_distinct_id` gets associated with the user.

Only one person is created, but the extra distinct ID adds noise when looking at the person. If we can confidently say that we never sent any events with this anonymous ID, either now or in the past, we can not send it.